zio-http/shared/src/main/scala/zio/http/Middleware.scala

```diff
@@ -337,7 +337,11 @@ object Middleware extends HandlerAspects {
     }
 
     def fromResource(resourcePrefix: String)(implicit trace: Trace): StaticServe[Any, Throwable] = make { (path, _) =>
-      Handler.fromResource(s"${resourcePrefix}/${path.dropLeadingSlash.encode}")
+      if (resourcePrefix.startsWith(".")) {
+        Handler.forbidden("Resource prefix cannot start with \".\"")
```

Comment:
I'm not really happy with this because we are passing off what should be a development-time (i.e., compile-time) error as a client-side HTTP error. Any ideas on how to handle this better?

Reply:
I think we could have either a string interpolator or make fromResource a macro, if we want to check this at compile time.
I think this would be easier if we would support only Scala 3.

But I don't think this is necessary. Just use an assert or throw an exception instead of returning the `Handler.forbidden` that way, it does not fail on request, but on start-up. Which should be good enough.

Also, I think the prefix should not be allowed to start with `~` or `/`. Maybe whitespace as well?

Reply:
You could use Refined to limit the string at compile time, but I agree with @987Nabil that it would be overkill.

Reply:
> Also, I think the prefix should not be allowed to start with `~` or `/`. Maybe whitespace as well?

Just `/` should be forbidden, but something like `/public/` should be fine. Does `~` means something in the context of resource paths?

Reply:
Maybe `Handler.die` should exist??

Reply:
@erikvanoosten `~` is the home path in Unix systems. And I would try to avoid some strange side effects. Maybe `~/` is what should be disallowed

Reply:
> @erikvanoosten `~` is the home path in Unix systems. And I would try to avoid some strange side effects. Maybe `~/` is what should be disallowed

`~` might resolve to something in the file system, but it has no meaning in resource paths, see https://docs.oracle.com/javase/8/docs/technotes/guides/lang/resources.html#res_names.

Reply:
> Maybe `Handler.die` should exist??

That sounds good, but I don't really know how what an implementation would look like. I took a look at `ZIO.orDie` to get some ideas, but that method is incredibly complicated.

Reply:
You can do, `Handler.fromZIO(ZIO.die(...))`. Composition FTW!